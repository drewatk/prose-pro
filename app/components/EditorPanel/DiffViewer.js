import React from "react";
import { connect } from "react-redux";

import { EditorState, convertFromRaw } from "draft-js";

import { Button } from "reactstrap";

import { UPDATE_EDITOR_STATE, SET_EDIT_STATE } from "app/actions/editor";

export const DiffViewer = props => {
  const { diffData, currentFile, gitAbstractions, dispatch } = props;
  // TODO: Bring blockquote & code styles from editor
  return (
    <div>
      <Button
        close
        style={{ cursor: "pointer", float: "right", marginRight: "50px" }}
        onClick={() =>
          gitAbstractions
            .switchToCurrentVersion(currentFile)
            .then(fileData =>
              dispatch({
                type: UPDATE_EDITOR_STATE,
                payload: EditorState.createWithContent(convertFromRaw(fileData))
              })
            )
            .then(() => dispatch({ type: SET_EDIT_STATE }))
            .catch(err =>
              console.error("Error in exiting checkpoint view: ", err)
            )
        }
      />
      <div style={{ height: "30px" }} />
      <div dangerouslySetInnerHTML={{ __html: diffData }} />
    </div>
  );
};

const mapStateToProps = ({
  editor: { diffData },
  currentFile,
  gitAbstractions
}) => ({ diffData, currentFile, gitAbstractions });

export default connect(mapStateToProps)(DiffViewer);
