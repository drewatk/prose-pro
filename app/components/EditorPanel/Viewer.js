import React from "react";
import { connect } from "react-redux";
import { stateToHTML } from "draft-js-export-html";

import { Button } from "reactstrap";

import { EditorState, convertFromRaw } from "draft-js";
import { UPDATE_EDITOR_STATE, SET_EDIT_STATE } from "app/actions/editor";

export const Viewer = props => {
  const { editorState, currentFile, gitAbstractions, dispatch } = props;
  // TODO: Bring blockquote & code styles from editor

  const html = stateToHTML(editorState.getCurrentContent());
  return (
    <div>
      <Button
        close
        style={{ cursor: "pointer", float: "right", marginRight: "50px" }}
        onClick={() =>
          gitAbstractions
            .switchToCurrentVersion(currentFile)
            .then(fileData =>
              dispatch({
                type: UPDATE_EDITOR_STATE,
                payload: EditorState.createWithContent(convertFromRaw(fileData))
              })
            )
            .then(() => dispatch({ type: SET_EDIT_STATE }))
            .catch(err =>
              console.error("Error in exiting checkpoint view: ", err)
            )
        }
      />
      <div dangerouslySetInnerHTML={{ __html: html }} />
    </div>
  );
};

const mapStateToProps = ({
  editor: { editorState },
  currentFile,
  gitAbstractions
}) => ({ editorState, currentFile, gitAbstractions });

const WithViewer = connect(mapStateToProps)(Viewer);

export default WithViewer;
