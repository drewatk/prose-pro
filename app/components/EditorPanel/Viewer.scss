.close-button {
  cursor: pointer;
  float: right;
  margin-right: 20px;
  color: var(--dark-color) !important;
}

.viewer {
  color: var(--dark-color);
}

.container {
  padding-right: 55px;
  padding-left: 30px;
}
