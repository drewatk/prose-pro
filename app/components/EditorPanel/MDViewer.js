import React from "react";
import { connect } from "react-redux";
import { stateToHTML } from "draft-js-export-html";

import Viewer from "./Viewer";

export const MDViewer = ({ editorState }) => {
  // TODO: Bring blockquote & code styles from editor
  const html = stateToHTML(editorState.getCurrentContent());
  return (
    <Viewer>
      <div dangerouslySetInnerHTML={{ __html: html }} />
    </Viewer>
  );
};

const mapStateToProps = ({ editor: { editorState } }) => ({ editorState });

export default connect(mapStateToProps)(MDViewer);
