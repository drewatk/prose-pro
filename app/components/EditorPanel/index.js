import React from "react";
import { connect } from "react-redux";
import ProseEditor from "app/components/ProseEditor";
import Viewer from "./Viewer";
import AutoSave from "app/components/AutoSave";

export const EditorPanel = props => {
  const { isEditable } = props;

  return (
    <React.Fragment>
      {isEditable && (
        <AutoSave>
          <ProseEditor />
        </AutoSave>
      )}
      {!isEditable && <Viewer />}
    </React.Fragment>
  );
};

EditorPanel.displayName = "EditorPanel";

const mapStateToProps = ({ editor: { isEditable } }) => ({ isEditable });
const WithEditorPanel = connect(mapStateToProps)(EditorPanel);

export default WithEditorPanel;
