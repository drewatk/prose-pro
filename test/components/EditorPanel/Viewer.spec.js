import React from "react";
import Enzyme, { shallow, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Viewer } from "app/components/EditorPanel/Viewer";
import { stateToHTML } from "draft-js-export-html";

Enzyme.configure({ adapter: new Adapter() });

jest.mock("draft-js-export-html", () => ({ stateToHTML: jest.fn() }));
jest.mock("draft-js");

describe("Viewer", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("matches snapshot", () => {
    const props = {
      editorState: { getCurrentContent: jest.fn() },
      dispatch: jest.fn(),
      currentFile: "currentfile",
      gitAbstractions: {
        switchToCurrentVersion: jest.fn().mockResolvedValue("fileData")
      }
    };
    const wrapper = shallow(<Viewer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("sets inner html", () => {
    const props = {
      editorState: { getCurrentContent: jest.fn() },
      dispatch: jest.fn(),
      currentFile: "currentfile",
      gitAbstractions: {
        switchToCurrentVersion: jest.fn().mockResolvedValue("fileData")
      }
    };
    const mockHTML = '<p id="test-id">Foobar!</p>';
    stateToHTML.mockReturnValueOnce(mockHTML);

    const wrapper = render(<Viewer {...props} />);

    expect(props.editorState.getCurrentContent).toHaveBeenCalledTimes(1);
    expect(stateToHTML).toHaveBeenCalledTimes(1);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
