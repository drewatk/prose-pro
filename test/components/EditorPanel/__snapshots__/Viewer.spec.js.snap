// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewer matches snapshot 1`] = `
<div>
  <Button
    close={true}
    color="secondary"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "float": "right",
        "marginRight": "50px",
      }
    }
    tag="button"
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
</div>
`;

exports[`Viewer sets inner html 1`] = `"<button type=\\"button\\" style=\\"cursor:pointer;float:right;margin-right:50px\\" class=\\"close\\" aria-label=\\"Close\\"><span aria-hidden=\\"true\\">&#xD7;</span></button><div><p id=\\"test-id\\">Foobar!</p></div>"`;
